sourcegraph/cody
# Cody Web: Fix cody web server endpoint configuration

Fixes https://linear.app/sourcegraph/issue/CODY-3782/cody-web-chat-fails-on-s2-with-not-authenticated-error

A few facts:
- Prior to this PR  we attempted to run auth even though we didn't have all settings propagated in activate extension from Cody Agent. 
- It worked previously because we had a slightly different flow of setting observation (recent PRs about singletons DI and auth setting changed this)
- We didn't notice this problem during those PR tests because we fell back on the dotcom instance in the logic, and Cody Web was conveniently connected to the dotcom instance by default. 
- So this PR seeds all vital information for Cody Web (auth token and importantly server endpoint before we hit auth flow) 

## Test plan
- Run Cody Web demo with s2 remote instance 
- Open the browser application dev tool tab and clear your index db storage 
- Refresh the page
- You should see no cors problems over standard gql requests in the network tab
- Chat should work properly
